addon/components/sequential-render.js

```diff
@@ -242,11 +242,22 @@ export default Component.extend({
 
   fetchData: task(function* () {
     let { queryParams, taskOptions } = getProperties(this, 'queryParams', 'taskOptions');
-    let content = yield get(this, 'fetchDataTask').perform(queryParams, taskOptions);
+    let content;
+    if (get(this, 'getData')) {
+      content = yield get(this, 'getData')();
+    } else {
+      content = yield get(this, 'fetchDataTask').perform(queryParams, taskOptions);
+    }
 
     set(this, 'content', content);
-    if (!(isNone(content) && get(this, 'renderPriority') === criticalRender)) {
-      this.updateRenderStates();
+    if (get(this, 'renderCallback')) {
+      if (!(isNone(content) && get(this, 'renderPriority') === criticalRender)) {
```

Comment:
Introducing check for isContentNone only if `renderCallback` is passed in, since there might be scenarios where only async operation needs to be performed without any return value expected.

Reply:
This might not solve for cases where its impossible to proceed to further priority items if the critical task returns empty.

Reply:
In scenarios where task returns empty, we will not pass `renderCallback` right ? , so the else block will take care of it right ?. Please correct me if i'm missing something

Reply:
We will not know when it might be empty.. renderCallback and this do not have any correlation.

Reply:
reverted to original in latest commit